src/test/java/bio/terra/service/dataset/DatasetIntegrationTest.java

```diff
@@ -481,22 +484,33 @@ static String writeListToScratch(
     return String.format("gs://%s/%s", blob.getBucket(), targetPath);
   }
 
-  static void addServiceAccountRoleToBucket(String bucket, String serviceAccount, Role role) {
+  static void addServiceAccountRoleToBucket(
+      String bucket, String serviceAccount, Role role, String userProject) {
     Storage storage = StorageOptions.getDefaultInstance().getService();
-    Policy iamPolicy = storage.getIamPolicy(bucket);
+    BucketSourceOption[] options =
+        Optional.ofNullable(userProject)
+            .map(p -> new BucketSourceOption[] {BucketSourceOption.userProject(p)})
+            .orElseGet(() -> new BucketSourceOption[0]);
+
+    Policy iamPolicy = storage.getIamPolicy(bucket, options);
     storage.setIamPolicy(
         bucket,
-        iamPolicy.toBuilder().addIdentity(role, Identity.serviceAccount(serviceAccount)).build());
+        iamPolicy.toBuilder().addIdentity(role, Identity.serviceAccount(serviceAccount)).build(),
+        options);
   }
 
-  static void removeServiceAccountRoleFromBucket(String bucket, String serviceAccount, Role role) {
+  static void removeServiceAccountRoleFromBucket(
+      String bucket, String serviceAccount, Role role, String userProject) {
     Storage storage = StorageOptions.getDefaultInstance().getService();
-    Policy iamPolicy = storage.getIamPolicy(bucket);
+    BucketSourceOption[] options =
+        Optional.ofNullable(userProject)
+            .map(p -> new BucketSourceOption[] {BucketSourceOption.userProject(p)})
+            .orElseGet(() -> new BucketSourceOption[0]);
```

Comment:
Since the `orElseGet` case is the same as an empty stream, you could use a stream here to avoid having an else branch, and avoid constructing an array on two different branches.
```suggestion
    BucketSourceOption[] options =
         Optional.ofNullable(userProject)
             .stream()
             .map(BucketSourceOption::userProject)
             .toArray(BucketSourceOption[]::new);
```